accounts.go

```diff
@@ -89,4 +108,33 @@ func (ac *account) PruneExpiredTxs() {
 	for _, sell := range *expiredSells {
 		sell.RollBack()
 	}
+
+	ac.AddSummaryItem("Finished expired TX cleanup")
+}
+
+type summaryItem struct {
+	loggedAt time.Time
+	message  string
+}
+
+func (ac *account) AddSummaryItem(s string) {
+	// Since ring.Do() always goes _forward_ we want to make sure the forward
+	// order of elements is newest -> oldest. This saves us a reverse after
+	// we convert the ring to a slice.
+	ac.summary = ac.summary.Prev()
+	ac.summary.Value = summaryItem{time.Now(), s}
+}
+
+// GetSummary returns a list of the user's most recent account activities,
+// sorted newest to oldest
+func (ac *account) GetSummary() []summaryItem {
+	s := make([]summaryItem, 0)
+
+	ac.summary.Do(func(node interface{}) {
+		if node != nil {
+			s = append(s, node.(summaryItem))
+		}
+	})
```

Comment:
This is pretty gross (lots of memory allocation!) but I can't find a nice way to shrink slices. I want to only return non-empty`summaryItem`s...